core/checksums/src/main/java/software/amazon/awssdk/checksums/internal/CrcChecksumProvider.java

```diff
@@ -65,14 +66,38 @@ public static SdkChecksum create() {
     static SdkChecksum createCrtCrc32C() {
         return CONSTRUCTOR_CACHE.getConstructor(CRT_CRC32C_CLASS_PATH).map(constructor -> {
             try {
-                return new CrcCloneOnMarkChecksum((Checksum) constructor.newInstance(), checksumToClone ->
-                    (Checksum) ((CRC32C) checksumToClone).clone());
-            } catch (ClassCastException | ReflectiveOperationException e) {
-                throw new IllegalStateException("Failed to instantiate " + JAVA_CRC32C_CLASS_PATH, e);
+                Checksum checksumInstance = (Checksum) constructor.newInstance();
+                return new CrcCloneOnMarkChecksum(checksumInstance);
+            } catch (ReflectiveOperationException e) {
```

Comment:
Curious, why is `ClassCastException` removed?

Reply:
Earlier we were doing (Checksum) , we no longer need this since CrcCloneOnMarkChecksum internally does check for Cloneable inetrface